src/utils/function.js

```diff
@@ -2,25 +2,34 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */
 
+// @flow
 import { findClosestFunction } from "./ast";
 import { correctIndentation } from "./indentation";
+import type { Source } from "../types";
+import type { Symbols } from "../reducers/ast";
 
-export function findFunctionText(line, source, symbols) {
+export function findFunctionText(
+  line: number,
+  source: Source,
+  symbols: ?Symbols
+): ?string {
   const func = findClosestFunction(symbols, {
+    sourceId: source.id,
     line,
     column: Infinity
   });
-  if (!func) {
+
+  if (!func || !source.text) {
     return null;
   }
 
   const { location: { start, end } } = func;
-  const lines = source.text.split("\n");
-  const firstLine = lines[start.line - 1].slice(start.column);
-  const lastLine = lines[end.line - 1].slice(0, end.column);
-  const middle = lines.slice(start.line, end.line - 1);
-  const functionText = [firstLine, ...middle, lastLine].join("\n");
-  const indentedFunctionText = correctIndentation(functionText);
+  const lines: string[] = source.text.split("\n");
+  const firstLine: string = lines[start.line - 1].slice(start.column);
+  const lastLine: string = lines[end.line - 1].slice(0, end.column);
+  const middle: string[] = lines.slice(start.line, end.line - 1);
+  const functionText: string = [firstLine, ...middle, lastLine].join("\n");
+  const indentedFunctionText: string = correctIndentation(functionText);
```

Comment:
do we need these for it to pass? they should be inferred...

Reply:
oh ok I can get rid of these. They are def unnecessary.